apps/web/screens/GenericOverview/GenericOverview.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react'
+import { Screen } from '@island.is/web/types'
+import { CustomNextError } from '@island.is/web/units/errors'
+import {
+  Breadcrumbs,
+  Stack,
+  Text,
+  Box,
+  Navigation,
+  GridColumn,
+  GridRow,
+  Button,
+} from '@island.is/island-ui/core'
+import { withMainLayout } from '@island.is/web/layouts/main'
+import { SidebarLayout } from '../Layouts/SidebarLayout'
+import { Document } from '@contentful/rich-text-types'
+import { GET_GENERIC_OVERVIEW_PAGE_QUERY } from '@island.is/web/screens/queries'
+import {
+  GetGenericOverviewPageQuery,
+  QueryGetGenericOverviewPageArgs,
+} from '@island.is/web/graphql/schema'
+import { LinkType, useLinkResolver } from '../../hooks/useLinkResolver'
+import NextLink from 'next/link'
+import { Image, renderHtml } from '@island.is/island-ui/contentful'
+
+interface GenericOverviewProps {
+  genericOverviewPage: GetGenericOverviewPageQuery['getGenericOverviewPage']
+}
+
+export const GenericOverview: Screen<GenericOverviewProps> = ({
+  genericOverviewPage: { title, intro, navigation, overviewLinks },
+}) => {
+  const { linkResolver } = useLinkResolver()
+
+  const generateUrl = (typename: LinkType, slug: Array<string>) => {
+    return typename.toLowerCase() === 'linkurl'
+      ? { href: slug[0] }
+      : {
+          ...linkResolver(typename, slug),
+        }
+  }
+
+  const renderNavigation = (isMobile: boolean) => {
+    return (
+      <Navigation
+        baseId={isMobile ? 'mobileNav' : 'desktopNav'}
+        isMenuDialog={isMobile}
+        activeItemTitle={navigation.title}
+        title={navigation.title}
+        items={[
+          ...navigation.menuLinks.map((item) => ({
+            title: item.title,
+            typename: item.link.type,
+            slug: [item.link.slug],
+          })),
+        ]}
+        renderLink={(link, { typename, slug }) => {
+          return (
+            <NextLink {...generateUrl(typename as LinkType, slug)} passHref>
+              {link}
+            </NextLink>
```

Comment:
no.. the link component is outdated (style and functionality wise), we just want to wrap the menulinks with nextjs when possible because we (web) are using nextjs